narwhal/consensus/src/consensus.rs

```diff
@@ -31,6 +35,114 @@ pub mod consensus_tests;
 /// The representation of the DAG in memory.
 pub type Dag = BTreeMap<Round, HashMap<AuthorityIdentifier, (CertificateDigest, Certificate)>>;
 
+#[derive(Default)]
+pub struct LeaderSwapTable {
+    pub good_nodes: Vec<AuthorityIdentifier>,
```

Comment:
Do these members have to be `pub`?

Reply:
Not really, I'll convert them to private